### PR TITLE
Sort values in AdminDynamicFieldScreenConfiguration

### PR DESCRIPTION
**Pull request by eyazi**
## Proposed change

In „Assignment of dynamic fields to screens“ (AdminDynamicFieldScreenConfiguration) screens and default columns get sorted by the key of the setting and not the value of the setting.

This PR adds proper sorting. The „sorting“ doesn't look that bad on the screenshots below, but if you add your own modules where the key differs from the screen name, the view gets confusing very fast.

## Type of change
- '1 - 🚀 feature'

## Additional information

Old:
Old

New:
New

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [x] Local ZnunyCodePolicy passed.(❕)
- [ ] Local UnitTests / Selenium passed.(❕)
- [ ] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by dennykorsukewitz** [comment]
Hi  @eyazi ,
thanks for your PR.

Please, keep your fork up-to-date with the upstream repository. 
[Syncing a fork 🚀  🆙 ](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork)

Regards :rocket:

**Comment by NiklasSchmitt** [comment]
Hi @eyazi,

thank you for your PR. I would like to accept and merge it but first you have to cleanup the code that the **ZnunyCodePolicy** and all other pipelines successes. Also, please [sync your fork](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) with the upstream repository.

Best regards,
Niklas Schmitt :coffee: :rocket:

File: Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm [hunk 1]
```diff
@@ -278,7 +278,7 @@ sub _ShowOverview {
 
     $LayoutObject->Block( Name => 'Overview' );
 
-    for my $DynamicFieldScreen ( sort keys %DynamicFieldScreens ) {
+    for my $DynamicFieldScreen ( sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} } keys %DynamicFieldScreens ) {
```
Comment (dennykorsukewitz) [inline Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm]:
```suggestion
    for my $DynamicFieldScreen ( sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} }
        keys %DynamicFieldScreens )
    {
```

File: Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm [hunk 2]
```diff
@@ -288,7 +288,7 @@ sub _ShowOverview {
         );
     }
 
-    for my $DefaultColumnsScreen ( sort keys %DefaultColumnsScreens ) {
+    for my $DefaultColumnsScreen ( sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} } keys %DefaultColumnsScreens ) {
```
Comment (dennykorsukewitz) [inline Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm]:
```suggestion
    for my $DefaultColumnsScreen ( sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} }
        keys %DefaultColumnsScreens )
    {
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]
```diff --git a/Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm b/Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm
index 0767c2f927..a292d6b14f 100644
--- a/Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm
+++ b/Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm
@@ -278,7 +278,9 @@ sub _ShowOverview {

     $LayoutObject->Block( Name => 'Overview' );

-    for my $DynamicFieldScreen ( sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} } keys %DynamicFieldScreens ) {
+    for my $DynamicFieldScreen ( sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} }
+        keys %DynamicFieldScreens )
+    {
         $LayoutObject->Block(
             Name => 'DynamicFieldScreenOverviewRow',
             Data => {
@@ -288,7 +290,9 @@ sub _ShowOverview {
         );
     }

-    for my $DefaultColumnsScreen ( sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} } keys %DefaultColumnsScreens ) {
+    for my $DefaultColumnsScreen ( sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} }
+        keys %DefaultColumnsScreens )
+    {
         $LayoutObject->Block(
             Name => 'DefaultColumnsScreenOverviewRow',
             Data => {
(END)
```

File: Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm [hunk 3]
```diff
@@ -278,7 +278,9 @@ sub _ShowOverview {
 
     $LayoutObject->Block( Name => 'Overview' );
 
-    for my $DynamicFieldScreen ( sort keys %DynamicFieldScreens ) {
+    for my $DynamicFieldScreen ( sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} }
+        keys %DynamicFieldScreens )
+    {
```
Comment (dennykorsukewitz) [inline Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm]:
```suggestion
    for my $DynamicFieldScreen (
        sort { $DynamicFieldScreens{$a} cmp $DynamicFieldScreens{$b} }
        keys %DynamicFieldScreens
        )
    {
```

File: Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm [hunk 4]
```diff
@@ -288,7 +290,9 @@ sub _ShowOverview {
         );
     }
 
-    for my $DefaultColumnsScreen ( sort keys %DefaultColumnsScreens ) {
+    for my $DefaultColumnsScreen ( sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} }
+        keys %DefaultColumnsScreens )
+    {
```
Comment (dennykorsukewitz) [inline Kernel/Modules/AdminDynamicFieldScreenConfiguration.pm]:
```suggestion
    for my $DefaultColumnsScreen (
        sort { $DefaultColumnsScreens{$a} cmp $DefaultColumnsScreens{$b} }
        keys %DefaultColumnsScreens
        )
    {
```

**Review by dennykorsukewitz — COMMENTED** [review]
@eyazi sorry for that.

Today the code policy is bit ...

**Review by dennykorsukewitz — APPROVED** [review]

**Review by NiklasSchmitt — APPROVED** [review]

**Pull request merged**